spec/index.bs

```diff
@@ -1493,6 +1497,103 @@ interface GPUTextureUsage {
+        1. Let |h| = 1.
+        1. Let |d| = 1.
+
+      1. Else if the {{GPUTextureDescriptor/dimension}} is "2d":
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = {{GPUTextureDescriptor/size}}.[=Extent3D/height=].
+        1. Let |d| = 1.
+
+      1. Else (the {{GPUTextureDescriptor/dimension}} is "3d"):
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = {{GPUTextureDescriptor/size}}.[=Extent3D/height=].
+        1. Let |d| = {{GPUTextureDescriptor/size}}.[=Extent3D/depth=].
+
+      1. Let |m| = the maximum value of |w|, |h|, and |d|.
+      1. Return one plus the greatest integral value of |x| for which 2^|x| <= |m|.
+
+    </dfn>
+  </div>
+
+  <div algorithm class=validusage>
+    <dfn abstract-op>createTexture Valid Usage</dfn>
+      Given a {{GPUDevice}} |this| and a {{GPUTextureDescriptor}} |descriptor|
+      the following validation rules apply:
+
+      1. |this| must be a [=valid=] {{GPUDevice}}.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be nonzero.
+      1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is "1d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be 1.
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. Else if |descriptor|.{{GPUTextureDescriptor/dimension}} is "2d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+
+      1. Else (|descriptor|.{{GPUTextureDescriptor/dimension}} is "3d"):
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. If |descriptor|.{{GPUTextureDescriptor/sampleCount}} > 1:
+
+        1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be 1.
```

Comment:
Also the texture must not have the STORAGE usage.

Reply:
And not be compressed.

Reply:
Can you point me to the place where we resolved about the STORAGE usage?

Reply:
It is in the "Sample Count" section of [this investigation](https://github.com/gpuweb/gpuweb/issues/513).